ibis-server/app/model/metadata.py

```diff
@@ -0,0 +1,357 @@
+            )(row)
+            for row in res["data"]
+        ]
+
+        unique_tables = {}
+        for row in response:
+            # generate unique table name
+            schema_table = f"{row['table_schema']}.{row['table_name']}"
+            # init table if not exists
+            if schema_table not in unique_tables:
+                table = {
+                    "name": schema_table,
+                    "description": "",
+                    "columns": [],
+                    "properties": {
+                        "schema": row["table_schema"],
+                        "catalog": row["table_catalog"],
+                        "table": row["table_name"],
+                    },
+                    "primaryKey": "",
+                }
+                unique_tables[schema_table] = CompactTable(**table)
+            # table exists, and add column to the table
+            column = {
+                "name": row["column_name"],
+                "type": row["data_type"],
+                "notNull": row["is_nullable"].lower() == "no",
+                "description": "",
+                "properties": {},
+            }
+            column = CompactColumn(**column)
+            unique_tables[schema_table].columns.append(column)
+
+        compact_tables: list[CompactTable] = list(unique_tables.values())
+        return compact_tables
+
+    @staticmethod
+    def get_bigquery_table_list_sql(connection_info):
+        dataset_id = connection_info.dataset_id
+        sql = f"""
+            SELECT 
+                c.table_catalog,
+                c.table_schema,
+                c.table_name,
+                c.column_name,
+                c.ordinal_position,
+                c.is_nullable,
+                c.data_type,
+                c.is_generated,
+                c.generation_expression,
+                c.is_stored,
+                c.is_hidden,
+                c.is_updatable,
+                c.is_system_defined,
+                c.is_partitioning_column,
+                c.clustering_ordinal_position,
+                c.collation_name,
+                c.column_default,
+                c.rounding_mode,
+                cf.description AS column_description, 
+                table_options.option_value AS table_description
+            FROM {dataset_id}.INFORMATION_SCHEMA.COLUMNS c 
+            JOIN {dataset_id}.INFORMATION_SCHEMA.COLUMN_FIELD_PATHS cf 
+                ON cf.table_name = c.table_name 
+                AND cf.column_name = c.column_name
+            LEFT JOIN {dataset_id}.INFORMATION_SCHEMA.TABLE_OPTIONS table_options
+                ON c.table_name = table_options.table_name
+            WHERE
+                cf.column_name = cf.field_path
+                AND NOT REGEXP_CONTAINS(cf.data_type, r'^(STRUCT|ARRAY<STRUCT)')
+            """
+        print(sql)
+        res = to_json(
+            DataSource.bigquery.get_connection(connection_info)
+            .sql(sql, dialect="bigquery")
```

Comment:
SQL is BigQuery SQL. We don't provide dialect for ibis converting sql from the specific dialect.